jobs/webcompat-kb/webcompat_kb/main.py

```diff
@@ -193,6 +210,26 @@ def fetch_bugs(self, params=None):
             self.bugs_fetch_completed = False
             return []
 
+    def filter_core_as_kb_bugs(self, other, kb_bugs_ids, site_reports_ids):
+        core_as_kb_bugs = []
+        for bug in other:
+            if CORE_AS_KB_KEYWORD in bug["keywords"]:
+                # Check if the core bug already has a kb entry and skip if so
+                if any(blocked_id in kb_bugs_ids for blocked_id in bug["blocks"]):
+                    continue
+
+                # Only store a breakage bug as it's the relation we care about
+                filtered_blocks = [
+                    blocked_id
+                    for blocked_id in bug["blocks"]
+                    if blocked_id in site_reports_ids
+                ]
+                if filtered_blocks:
```

Comment:
For `Web Compatibility :: Knowledge Base` bugs do we only include entries with linked site reports? It seems to me that it might make sense to include ones even if there aren't any current site reports, even if only so we can write queries to find mislabeled bugs. In any case we should probably match the behaviour in both cases.

Reply:
Yeah we include all bugs from `Web Compatibility :: Knowledge Base`, even those without reports. Same with any bug that is not in Web Compatibility component, but has `webcompat:` keyword. They all end up in bugzilla_bugs table (these are the filters: https://github.com/mozilla/docker-etl/blob/main/jobs/webcompat-kb/webcompat_kb/main.py#L28 for the api request).

So all bugs from this list are imported into that table (even though they don't have `webcompat:platform-bug` keyword in particular):
https://bugzilla.mozilla.org/buglist.cgi?v1=Web%20Compatibility&classification=Client%20Software&classification=Developer%20Infrastructure&classification=Components&classification=Server%20Software&classification=Other&f2=keywords&f1=product&resolution=---&v2=webcompat%3A&n1=1&o2=regexp&query_format=advanced&list_id=17052706&o1=equals

In that list there are [3 bugs](https://bugzilla.mozilla.org/buglist.cgi?v1=Web%20Compatibility&classification=Client%20Software&classification=Developer%20Infrastructure&classification=Components&classification=Server%20Software&classification=Other&f2=keywords&f1=product&resolution=---&v2=webcompat%3Aplatform-bug&n1=1&o2=regexp&query_format=advanced&list_id=17052706&o1=equals) with `webcompat:platform-bug` keyword and [only one has a linked breakage report at the moment](https://bugzilla.mozilla.org/show_bug.cgi?id=1886820), so it will also have a relation record in the `breakage_reports` table as if it was a knowledge base bug (these are the changes in this PR).

| Row  | knowledge_base_bug | breakage_bug |
|:----:|:------------------:|--------------|
|   1   | 1886820            | 1896383      |
	

Does that sound reasonable?

Reply:
I think we should import all three as kb entries, so that we can easily do a search like "show me all the kb entries with no linked site reports".

Reply:
Talked to James on slack and decided to crate a separate table to store ids of kb entries

Reply:
I've made the changes, could you please take another look @jgraham?